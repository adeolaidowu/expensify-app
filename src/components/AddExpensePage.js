import React from 'react';
import { connect } from 'react-redux';
import { startAddExpense } from '../actions/expenses';
import ExpenseForm from './ExpenseForm';

export class AddExpensePage extends React.Component {
  onSubmit = (expense) => {
    //props.dispatch(addExpense(expense));
    this.props.startAddExpense(expense);
    this.props.history.push('/');
  }
  render() {
    return (
      <div>
        <h1>Add Expense</h1>
        <ExpenseForm
          onSubmit={this.onSubmit}
        />
      </div>
    );
  }
};

// same as above except using functional component
// const AddExpensePage = (props) => {
//   const onSubmit = (expense) => {
//     //props.dispatch(addExpense(expense));
//     props.addExpense(expense);
//     props.history.push('/');
//   }
//   return (
//     <div>
//       <h1>Add Expense</h1>
//       <ExpenseForm
//         onSubmit={onSubmit}
//       />
//     </div>
//   );
// }

const mapDispatchToProps = (dispatch) => (
  {
    startAddExpense: (expense) => dispatch(startAddExpense(expense))
  }
)

export default connect(undefined, mapDispatchToProps)(AddExpensePage);