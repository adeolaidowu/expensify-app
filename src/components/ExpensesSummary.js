import React from 'react';
import { connect } from 'react-redux';
import getVisibleExpenses from '../selectors/expenses';
import getTotalExpenses from '../selectors/expensesTotal';
import numeral from 'numeral';

 export const ExpensesSummary = (props) => {
  const numberOfExpenses = props.expenses.length;
  const expensesTotal = getTotalExpenses(props.expenses);
  const formattedTotal = numeral(expensesTotal / 100).format('$0,0.00');
  const expenseWord = numberOfExpenses === 1 ? 'expense' : 'expenses';
  return (
    <div>
      <h1>Viewing {numberOfExpenses} {expenseWord} totalling {formattedTotal} </h1>
    </div>
  );
};

const mapStateToProps = (state) => ({
  expenses: getVisibleExpenses(state.expenses, state.filters)
});

export default connect(mapStateToProps)(ExpensesSummary);