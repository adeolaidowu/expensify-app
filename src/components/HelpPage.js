import React from 'react';

const HelpPage = () => (
  <div>
    What do you need help with?
  </div>
);

export default HelpPage;