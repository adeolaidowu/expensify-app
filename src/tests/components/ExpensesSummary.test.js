import React from 'react';
import { shallow } from 'enzyme';
import { ExpensesSummary } from '../../components/ExpensesSummary'
import expenses from '../fixtures/expenses';

test('should render ExpensesSummary correctly with 1 expense', () => {
  const wrapper = shallow(<ExpensesSummary expenses={[expenses[2]]}/>);
  expect(wrapper).toMatchSnapshot();
});

test('should render ExpensesSummary correctly with multiple expenses', () => {
  const wrapper = shallow(<ExpensesSummary expenses={expenses} />);
  expect(wrapper).toMatchSnapshot();
});