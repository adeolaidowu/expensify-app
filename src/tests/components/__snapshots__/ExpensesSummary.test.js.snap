// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ExpensesSummary correctly with 1 expense 1`] = `
<div>
  <h1>
    Viewing 
    1
     
    expense
     totalling 
    $45.00
     
  </h1>
</div>
`;

exports[`should render ExpensesSummary correctly with multiple expenses 1`] = `
<div>
  <h1>
    Viewing 
    3
     
    expenses
     totalling 
    $57.90
     
  </h1>
</div>
`;
